vignettes/wrapping-apis.Rmd

```diff
@@ -213,92 +224,130 @@ faker_person <- function(gender = NULL, birthday_start = NULL, birthday_end = NU
+There are three basic steps to this process:
 
-I'll start by tackling the first problem because otherwise there's no way for me to show how the API works in this vignette 😃.
-We'll come back to the second at the very end of this section, because it's easiest to tackle once we have a function in place.
+1.  You create an **encryption** key with `secret_make_key()` that is used to scramble and descramble secrets using symmetric cryptography:
 
-### Encrypting secrets
+    ```{r}
+    key <- secret_make_key()
+    key
+    ```
 
-httr2 provides the `secret_decrypt()` and `secret_decrpyt()` functions that use asymmetric cryptography.
-To use them you need to first create a "key" that's used to lock and unlock the secrets.
-The best key is random because there's no way for anyone to guess it.
-httr2 provides `secret_make_key()` to make a secure key for you:
+    (Note that `secret_make_key()` uses a cryptographically secure random number generator provided by OpenSSL; it is not affected by R's RNG settings, and there's no way to make it reproducible.)
 
-```{r}
-secret_make_key()
-```
+2.  You scramble your secrets with `secret_encrypt()` and store the resulting text directly in the source code of your package:
 
-(Don't copy and paste this key; run `secret_make_key()` yourself to make your own personal key.)
+    ```{r}
+    secret <- secret_encrypt("this is a secret message", key)
+    secret
+    ```
 
-Now you need to make this key available in the places were you want to unlock the secrets.
-httr2 provides tools to retrieve keys from environment variables because this is a common way of communicating these sort of secrets.
-So start by opening your `.Renviron` (e.g. with `usethis::edit_r_environ()`) and add a line like:
+3.  When needed, you descramble the the secret using `secret_decrypt()`:
+
+    ```{r}
+    secret_decrypt(secret, key)
+    ```
 
-    YOURPACKAGE_KEY=key_that_you_generated_above
+### Package secrets
 
-Another place you'll probably want to provide the key is in your Github actions.
-This is a two step process:
+You can create an number of encryption keys, but I highly recommend that you create one key per package, which I'll call the **package** key.
+In this section, I'll show you how to store that key so that you (and your automated tests) can use it, but no one else can.
+
+httr2 is built around the notion that this key should live in an environment variable.
+So the first step is to make your package key available on your local development machine by adding a line to your your user-level `.Renviron` (which you can easily open with `usethis::edit_r_environ()`):
+
+    YOURPACKAGE_KEY=key_you_generated_with_secret_make_key
+
+```{r, include = FALSE}
+Sys.setenv(YOURPACKAGE_KEY = secret_make_key())
+```
+
+Now (after you restart R), you'll be able to take advantage of a special `secret_encrypt()` and `secret_decrypt()` feature: the encryption key can be the name of an environment variable.
```

Comment:
```suggestion
Now (after you restart R), you'll be able to take advantage of a special `secret_encrypt()` and `secret_decrypt()` feature: the `key` argument can be the name of an environment variable, instead of the encryption key itself.
In fact, this is most natural usage.
```

Reply:
I'm not totally happy with "natural" but I want to convey "most intended", "most planned for", "should be most common".